src/dune_console/dune_console.ml

```diff
@@ -189,33 +248,41 @@ module Backend = struct
       let open T in
       let last = ref (Unix.gettimeofday ()) in
       let frame_rate = 1. /. 60. in
+      let cleanup () =
+        state.finished <- true;
+        Base.finish ();
+        Condition.broadcast finish_cv;
+        Mutex.unlock mutex
+      in
       try
+        Base.start ();
         while true do
           Mutex.lock mutex;
-          while not (Queue.is_empty state.messages) do
-            Base.print_user_message (Queue.pop_exn state.messages)
-          done;
-          Base.set_status_line state.status_line;
-          flush stderr;
           let finish_requested = state.finish_requested in
           if finish_requested then raise_notrace Exit;
+          Base.render state;
```

Comment:
you've reintroduced the flushing bug again

Reply:
fixed